src/DataService/NotificationListener.php

```diff
@@ -95,16 +103,22 @@ public function sync()
     /**
      * Listen for notifications.
      *
+     * This method always filters to the object type defined in the discovered
+     * data service passed in the constructor, such as 'Media'.
+     *
+     * @see  https://docs.theplatform.com/help/media-media-data-service-api-reference
+     *
      * @todo Add support for a configurable timeout?
-     * @todo Convert the return to an ObjectList.
      *
-     * @see \Lullabot\Mpx\DataService\NotificationListener::sync
+     * @see  \Lullabot\Mpx\DataService\NotificationListener::sync
+     * @see  \Lullabot\Mpx\DataService\Notification
      *
-     * @param int $since The last sync ID.
+     * @param int $since   The last sync ID.
+     * @param int $maximum (optional) The maximum number of notifications to return. Defaults to 500.
      *
-     * @return \GuzzleHttp\Promise\PromiseInterface A promise returning a list of notified objects.
+     * @return \GuzzleHttp\Promise\PromiseInterface A promise returning an array of Notifications.
```

Comment:
Do you mean "returning an array of Notification objects"?

Reply:
Right, I meant that by capitalizing Notifications, but that's totally not obvious.